joblib/format_stack.py

```diff
@@ -196,6 +196,10 @@ def format_records(records):   # , print_globals=False):
             # keep the original file string.
             pass
         link = file
+
+        if file.endswith('.pyc'):
+            file = file[:-4] + '.py'
```

Comment:
> Can you put these two lines above the link = file line?

That's where the added lines were meant to be, I must have accidentally swapped the places.

Also, I am still working on the test, so bear with me please. :)